src/security/builtin_plugins/cryptographic/src/crypto_key_factory.c

```diff
@@ -89,8 +89,10 @@ calculate_kx_keys(
   unsigned char *kx_master_salt, *kx_master_sender_key;
   size_t shared_secret_size = get_secret_size_from_secret_handle(shared_secret);
   unsigned char hash[SHA256_DIGEST_LENGTH];
-  size_t concatenated_bytes1_size = DDS_SECURITY_AUTHENTICATION_CHALLENGE_SIZE * 2 + sizeof(KXSALTCOOKIE);
-  size_t concatenated_bytes2_size = DDS_SECURITY_AUTHENTICATION_CHALLENGE_SIZE * 2 + sizeof(KXKEYCOOKIE);
+  size_t KXKEYCOOKIE_SIZE = strlen(KXKEYCOOKIE);
```

Comment:
While correct, I think it makes more sense to add `#define KXKEYCOOKIE_SIZE (sizeof(KXKEYCOOKIE)-1)` immediately after the definition of `KXKEYCOOKIE` (and same for `KXSALTCOOKIE`).